src/test/lrsql/admin/protocol_test.clj

```diff
@@ -155,6 +183,49 @@
         (is (-> (adp/-ensure-account-oidc lrs username bad-issuer)
                 :result
                 (= :lrsql.admin/oidc-issuer-mismatch-error)))))
+    (testing "actor-delete"
+      (let [stmts [stmt-0 stmt-1 stmt-2 stmt-3]
+            ifis (->> (conj stmts (stmt-3 "object"))
+                      (map #(ua/actor->ifi (% "actor"))))
+            get-stmt-#s (fn []
+                          (reduce (fn [m actor-ifi]
+                                    (assoc m actor-ifi
+                                           (count (:statements (:statement-result (lrsp/-get-statements lrs auth-ident {:actor-ifi actor-ifi} []))))))
+                                  {} ifis))]
+        (lrsp/-store-statements lrs auth-ident stmts [])
+
+        (assert (every? #(> % 0) (vals (get-stmt-#s))))
+        (doseq [ifi ifis]
+          (adp/-delete-actor lrs {:actor-ifi ifi}))
+        (is (every? #(= % 0) (vals (get-stmt-#s))))))
+    (let [stmt->ifi #(ua/actor->ifi (% "actor"))
+          count-of-actor (fn [actor-ifi] (-> (lrsp/-get-statements lrs auth-ident {:actor-ifi actor-ifi} []) :statement-result :statements count))
+          child-ifi (stmt->ifi (stmt-3 "object"))
+          parent-ifi (stmt->ifi stmt-3)]
+      (testing "delete-actor correctly deletes statements that are parent to actor (sub)statements"
```

Comment:
So these tests are good progress, but I would like to see tests for everything that can be potentially deleted when you delete an actor. So that would mean:

- [x] Actor -> Statement
- [x] Actor -> Statement -> Parent Statement
- [x] Actor -> Statement -> Child Statement
- [ ] Actor -> Statement -> Activities
- [ ] Actor -> Statement -> Attachments
- [ ] Actor -> State Document
- [ ] Actor -> Agent Profile Document

Reply:
I don't see a test for the actor itself? (Which I find odd since wouldn't that be the _simplest_ thing to test?)

Reply:
There is one now.